int32 cmd
float64 pos1
float64 pos2
float64 pos3
float64 pos4
int32 action_time
---
int32 is_done
